src/Simulation/Simulators/SparseSimulator/Native/SparseSimulator.h

```diff
@@ -0,0 +1,846 @@
+#pragma once
```

Comment:
(Up to you)
This is a C++ file. 

You may want to change the extension from .h to .hpp.
You may want to break up the file into .hpp and .cpp.